pystiche/papers/ulyanov_et_al_2016/data.py

```diff
@@ -0,0 +1,293 @@
+            urljoin(base_ulyanov_suppl_texture, "red-peppers256.o.jpg"),
+            md5="16371574a10e0d10b88b807204c4f546",
+        ),
+    }
+    base_johnson = (
+        "https://raw.githubusercontent.com/jcjohnson/fast-neural-style/master/images/"
+    )
+    style_base_johnson = urljoin(base_johnson, "styles/")
+
+    base_ulyanov_suppl_style = "https://raw.githubusercontent.com/DmitryUlyanov/texture_nets/texture_nets_v1/supplementary//stylization_models/"
+    style_images = {
+        "candy": DownloadableImage(
+            urljoin(style_base_johnson, "candy.jpg"),
+            md5="00a0e3aa9775546f98abf6417e3cb478",
+        ),
+        "the_scream": DownloadableImage(
+            urljoin(style_base_johnson, "the_scream.jpg"),
+            md5="619b4f42c84d2b62d3518fb20fa619c2",
+        ),
+        "Jean Metzinger": DownloadableImage(
+            "https://upload.wikimedia.org/wikipedia/commons/c/c9/Robert_Delaunay%2C_1906%2C_Portrait_de_Metzinger%2C_oil_on_canvas%2C_55_x_43_cm%2C_DSC08255.jpg",
+            title="Portrait of Jean Metzinger",
+            author="Jean Metzinger",
+            date="1906",
+            license=PublicDomainLicense(1906),
+            md5="3539d50d2808b8eec5b05f892d8cf1e1",
+            file="jean_metzinger.jpg",
+        ),
+        "mosaic": DownloadableImage(
+            urljoin(base_ulyanov_suppl_style, "mosaic.jpg"),
+            md5="4f05f1e12961cebf41bd372d909342b3",
+        ),
+        "pleades": DownloadableImage(
+            urljoin(base_ulyanov_suppl_style, "pleades.jpg"),
+            md5="6fc41ac30c2852a5454a0ead2f479dc9",
+        ),
+        "starry": DownloadableImage(
+            urljoin(base_ulyanov_suppl_style, "starry.jpg"),
+            md5="c6d94f7962466b2e80a64ae82523242a",
+        ),
+        "turner": DownloadableImage(
+            urljoin(base_ulyanov_suppl_style, "turner.jpg"),
+            md5="7fdd9603a5182dcef23d7fb1c5217888",
+        ),
+    }
+    return DownloadableImageCollection(
+        {**texture_images, **content_images, **style_images},
+        root=root,
+        download=download,
+        overwrite=overwrite,
+    )
+
+
+def ulyanov_et_al_2016_dataset(
+    root: str, impl_mode: str = "paper", transform: Optional[Transform] = None,
+):
+    if transform is None:
+        transform = ulyanov_et_al_2016_content_transform(impl_mode=impl_mode)
+    return ImageFolderDataset(root, transform=transform)
+
+
+def ulyanov_et_al_2016_batch_sampler(
+    data_source: Sized,
+    impl_mode: str = "paper",
+    mode: str = "texture",
+    num_batches=None,
+    batch_size=None,
+) -> FiniteCycleBatchSampler:
+    if num_batches is None:
+        if impl_mode == "texturenetsv1":
+            num_batches = 3000 if mode == "style" else 1500
+        elif impl_mode == "paper":
+            num_batches = 2000
+        elif impl_mode == "master":
+            num_batches = 50000
+        else:
+            raise NotImplementedError
```

Comment:
A [`NotImplementedError`](https://docs.python.org/3/library/exceptions.html#NotImplementedError) implies that this method is abstract, i.e. should be overwritten in a derived class. Thus, a [`RuntimeError`](https://docs.python.org/3/library/exceptions.html#RuntimeError) would be a better fit here.